gnome-initial-setup/gis-assistant.c

```diff
@@ -103,17 +102,38 @@ gis_assistant_switch_to (GisAssistant          *assistant,
   gtk_stack_set_visible_child (GTK_STACK (priv->stack), GTK_WIDGET (page));
 }
 
+static inline gboolean
+should_show_page (GList *l)
+{
+  return l != NULL && gtk_widget_get_visible (GTK_WIDGET (l->data));
+}
+
+static GisPage *
+find_next_page (GisPage *page)
+{
+  GList *l = page->assistant_priv->link->next;
+  while (!should_show_page (l)) {
+    l = l->next;
+  }
+  return GIS_PAGE (l->data);
```

Comment:
What happens if `l == NULL` here due to running out of pages to show?

Reply:
Indeed! The simple answer is that this doesn't happen, because the forward button is only shown if we are not on the last page (and `should_show_page` returns TRUE for the last page). The same bomb appears in g-i-s upstream so I suppose I'll go fix it there.

Reply:
Sounds good

Reply:
https://bugzilla.gnome.org/attachment.cgi?id=368925&action=edit